File: BACKEND/facturador_backend/src/main/java/com/bap/api/services/api/ApiNotaFiscalCreditoDebitoService.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.bap.api.services.api;

import com.bap.api.configurate.AbstractEntity;
import com.bap.api.model.api.ApiNotaFiscalCreditoDebito;
import java.util.List;

/**
 *
 * @author ruth //
 */
public interface ApiNotaFiscalCreditoDebitoService extends AbstractEntity<ApiNotaFiscalCreditoDebito> {

    List<ApiNotaFiscalCreditoDebito> leerPorLogin(String login, Long codigoTipoDocumentoFiscal, Long codigoTipoDocumentoSector, Long codigoTipoModalidad);

    long verificaRepiteNotaFiscalCreditoDebitoPorSucursal(Long idSucursal, Long tipoDocumentoFiscal, Long tipoDocumentoSector, Long tipoModalidad, Long numeroNotaCreditoDebito);

    long verificaRepiteNotaFiscalCreditoDebitoPorPuntoVenta(Long idPuntoVenta, Long tipoDocumentoFiscal, Long tipoDocumentoSector, Long tipoModalidad, Long numeroNotaCreditoDebito);

    ApiNotaFiscalCreditoDebito findNotaFiscalCreditoDebitoPorSucursal(Long numeroNotaCreditoDebito, String numeroAutorizacionCuf, Long idSucursal, String fechaEmision);

    ApiNotaFiscalCreditoDebito findNotaFiscalCreditoDebitoPorPuntoVenta(Long numeroNotaCreditoDebito, String numeroAutorizacionCuf, Long idPuntoVenta, String fechaEmision);
}

File: FRONTEND/facturador-frontend/src/services/ApiNotaFiscalCreditoDebitoService.js
import Api from '@/services/Api'

export default { 
    //ApiNotaFiscalCreditoDebito
    leerPorLogin(login, codigoDocumentoFiscal, codigoDocumentoSector, codigoTipoModalidad){
        console.log("login", login);
        console.log("codigoDocumentoFiscal", codigoDocumentoFiscal);
        console.log("codigoDocumentoSector", codigoDocumentoSector);
        console.log("codigoTipoModalidad", codigoTipoModalidad);
        return Api().get(`/api/apiNotaFiscalCreditoDebito/leerPorLogin/${login}/${codigoDocumentoFiscal}/${codigoDocumentoSector}/${codigoTipoModalidad}`)
    },
}
